src/main/java/de/tum/in/www1/artemis/config/migration/MigrationService.java

```diff
@@ -0,0 +1,158 @@
+        return entryClassMap.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, entry ->
+        // We have to manually autowire the components here
+        (MigrationEntry) beanFactory.autowire(entry.getValue(), AutowireCapableBeanFactory.AUTOWIRE_CONSTRUCTOR, true), (prev, next) -> next, TreeMap::new));
+    }
+
+    /**
+     * Checks the given changelog for integrity. It must have distinct keys, distinct date strings and the date strings must be in order.
+     * It is set public to allow an external integrity check without starting the actual application and therefore migration.
+     * All occurring errors will be logged but duplicate errors are not logged.
+     *
+     * @param entryMap A changelog in form of an entryMap that should be checked
+     * @return True if the check was successful, otherwise false
+     */
+    public boolean checkIntegrity(SortedMap<Integer, MigrationEntry> entryMap) {
+        log.info("Starting migration integrity check");
+        boolean passed = true;
+        Map<Integer, MigrationEntry> brokenInstances = entryMap.entrySet().stream()
+                .filter(entry -> StringUtils.isEmpty(entry.getValue().date()) || StringUtils.isEmpty(entry.getValue().author()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        if (!brokenInstances.isEmpty()) {
+            passed = false;
+            log.error(getClass().getSimpleName() + " corrupted. Migration entries not properly set up.");
+            brokenInstances.forEach((key, value) -> log.error("Entry " + value.getClass().getSimpleName() + " has not all information methods defined."));
+            log.info("Please refer to the documentation on how to set up migration entries.");
+        }
+        List<MigrationEntry> entryList = entryMap.values().stream().toList();
+        if (!entryList.isEmpty()) {
+            int startIndex = 1;
+            MigrationEntry baseEntry = entryList.get(0);
+            // Make sure the base date is not null. If it is, it was already caught and logged above.
+            while (StringUtils.isEmpty(baseEntry.date()) && startIndex < entryList.size()) {
+                baseEntry = entryList.get(startIndex);
+                startIndex++;
+            }
+            // Go through the list of registered entries and make sure that every date is smaller than the date of it's successor.
+            for (int i = startIndex; i < entryList.size(); i++) {
+                MigrationEntry entry = entryList.get(i);
+                if (!StringUtils.isEmpty(entry.date()) && baseEntry.date().compareTo(entry.date()) >= 0) {
+                    log.error(getClass().getSimpleName() + " corrupted. Invalid date order detected. " + entry.date() + " (" + entry.getClass().getSimpleName()
+                            + ") should come before " + baseEntry.date() + " (" + baseEntry.getClass().getSimpleName() + ")");
```

Comment:
```suggestion
                    log.error("{} corrupted. Invalid date order detected. {} ({}) should come before {} ({})", getClass().getSimpleName(), entry.date(), entry.getClass().getSimpleName(), baseEntry.date(), baseEntry.getClass().getSimpleName());
```
Using the built-in formatting for the log messages is used in most other places. I also think it looks a lot cleaner compared to concatenating strings.
(Comment applies to the other log statements as well).

Reply:
Good suggestion!